File: src/main/java/com/github/qky666/selenidepom/java/RequiredError.java
package com.github.qky666.selenidepom.java;

import com.codeborne.selenide.ex.UIAssertionError;

import javax.annotation.CheckReturnValue;
import java.util.List;

/**
 * Error thrown when a {@link Required} element is not found.
 */
public class RequiredError extends UIAssertionError {
    /**
     * Constructor.
     *
     * @param errors List of Throwable errors found while looking for {@link Required} elements.
     */
    @CheckReturnValue
    public RequiredError(List<Throwable> errors) {
        super("Required elements not found in page: " + errors.size() + " errors. See suppressed errors");
        errors.forEach(this::addSuppressed);
    }
}

File: src/main/kotlin/com/github/qky666/selenidepom/kotlin/RequiredError.kt
package com.github.qky666.selenidepom.kotlin

import com.codeborne.selenide.ex.UIAssertionError

/**
 * Error thrown when a [Required] element is not found.
 */
class RequiredError(errors: List<Throwable>) :
    UIAssertionError("Required elements not found in page: ${errors.size} errors. See suppressed errors") {
    init {
        errors.forEach { this.addSuppressed(it) }
    }
}
